content/post/081-js-libp2p-0.27.md

```diff
@@ -0,0 +1,164 @@
+---
+date: 2020-01-29
+url: 2020-01-29-js-libp2p-0-27
+title: js-libp2p 0.27 released
+author: Jacob Heun
+---
+
+> The Async / Await Refactor and a whole lot more!
+
+# 🔦 Highlights
+
+## 📜 Improved docs
+
+We've done an overhaul of our docs to make libp2p easier to use. Among other docs in the new [doc][docs] folder, you can find a full list of exposed methods in the [API.md][api], and a guide on how to configure libp2p in [CONFIGURATION.md][config]. We've also created a [Getting Started guide][getting_started] for anyone starting out with libp2p.
+
+## ⌚️ Async/Await instead of Callbacks
+
+All callback APIs have been change to be async / await compliant. See the [API.md][api] readme for detailed usage. When migrating, you can leverage the [migration guide][migration] to see samples on some of the common migrations you may need to make.
+
+## 🚰 Streaming Iterables instead of Pull Streams
+
+Now that readable streams are async iterable, we can leverage [Streaming Iterables][streaming_iterables] instead of Pull Streams to greatly simplify the internal stream logic of libp2p. Among other things, this makes debugging streams much easier. You can check out the [it-awesome repo](https://github.com/alanshaw/it-awesome) for a list of an increasing number of modules built for the streaming iterables ecosystem. This also includes modules to convert to and from pull streams if you need to refactor your applications over time. If you're having trouble migrating, please feel free to reach out on the [discuss forums][discuss]!
+
+## 📞 Clearer Connections
+
+We've created a whole new [Connection Interface][connection]! Creating multiple streams off of a single connection is now much clearer, and every stream created is tracked in the Connection. This makes it much easier to keep track of every open stream, which greatly empowers resource management in js-libp2p.
+
+```js
+// Was
+libp2p.dialProtocol(remotePeerInfo, protocol, (error, stream) => { })
+
+// Now
+const connection = await libp2p.dial(remotePeerInfo)
+const { stream, protocol } = connection.newStream(protocols)
```

Comment:
This is async according to https://github.com/libp2p/js-interfaces/blob/master/src/connection/connection.js#L128

```suggestion
const { stream, protocol } = await connection.newStream(protocols)
```